main.go

```diff
@@ -37,11 +37,13 @@ import (
 var (
 	localMasterURL  string
 	localKubeconfig string
+	localEngine     string
 )
 
 func init() {
 	flag.StringVar(&localKubeconfig, "kubeconfig", "", "Path to kubeconfig of local cluster. Only required if out-of-cluster.")
 	flag.StringVar(&localMasterURL, "master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
+	flag.StringVar(&localEngine, "engine", "strongswan", "The IKE implementation to use: strongswan or libreswan.")
```

Comment:
How about we put this in ipsec.go instead of passing to ipsec.NewEngine? Also name `ipsec` instead of `engine` for clarity.

Longer term, I think we should consider using [Go plugins ](https://code.tutsplus.com/tutorials/writing-plugins-in-go--cms-29101) instead of passing CLI args to make it cleaner and easier to support other cable engine implementations.

Reply:
Or even ipsecDriver as the name?

Reply:
ipsecdriver ? :-)

I think that's more user friendly/clear.